java-frontend/src/main/java/org/sonar/java/se/xproc/ExceptionalCheckBasedYield.java

```diff
@@ -138,6 +148,28 @@ public int hashCode() {
       .hashCode();
   }
 
+  public Set<List<JavaFileScannerContext.Location>> exceptionFlows() {
+    Set<List<JavaFileScannerContext.Location>> flows = FlowComputation.flow(node.parent(), svCausingException, domains(node.programState.getConstraints(svCausingException)));
+    Tree syntaxTree = node.programPoint.syntaxTree();
+    ImmutableSet.Builder<List<JavaFileScannerContext.Location>> flowBuilder = ImmutableSet.builder();
+
+    for (List<JavaFileScannerContext.Location> flow : flows) {
+      List<JavaFileScannerContext.Location> newFlow = ImmutableList.<JavaFileScannerContext.Location>builder()
+        .add(new JavaFileScannerContext.Location(exceptionType().name() + " is thrown.", syntaxTree))
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') NullPointerException might be thrown as 'exceptionType' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)